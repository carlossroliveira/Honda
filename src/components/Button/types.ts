// -------------------------------------------------
// Packages
// -------------------------------------------------
import { ButtonHTMLAttributes } from 'react';
import { IconBaseProps } from 'react-icons';
// -------------------------------------------------
// Types
// -------------------------------------------------

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  text: string;
  icon?: IconBaseProps;
  onClick?: () => void;
}
