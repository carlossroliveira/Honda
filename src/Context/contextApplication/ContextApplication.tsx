/* eslint-disable @typescript-eslint/no-explicit-any */
// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { createContext, useContext, useState } from 'react';
import photo from '../../assets/GroupModal.svg';
import Swal from 'sweetalert2';
// -------------------------------------------------
// Hooks
// -------------------------------------------------
import { useFetch } from '../../hooks/useFetch';
// -------------------------------------------------
// Types
// -------------------------------------------------
import {
  IContextApplication,
  IContextApplicationProvider,
  IProductProps,
} from './types';

const Context = createContext({} as IContextApplication);

export const ThemeProviderApplication = ({
  children,
}: IContextApplicationProvider): JSX.Element => {
  const { data } = useFetch<IProductProps[]>('http://localhost:5000/info');

  const [array, setArray] = useState<any>([]);
  const [values, setValues] = useState<number>(0);

  if (values !== 4) {
    if (array.some((x: any) => array.indexOf(x) !== array.lastIndexOf(x))) {
      Swal.fire({
        text: `You've already added ${array.name} to your basket. Do you want to remove it ?`,
        showCancelButton: true,
        confirmButtonColor: '#21E7C5',
        confirmButtonText: 'Yes',
        cancelButtonText: 'No keep it',
        cancelButtonColor: '#C6C9D6',

        imageUrl: photo,
        imageWidth: 200,
        imageHeight: 200,
        imageAlt: 'Custom image',
      }).then(() => {
        array.pop();
      });
    }
  }

  const handleValue = () => {
    if (values < 4) {
      setValues(values + 1);
    } else {
      const Toast = Swal.mixin({
        toast: true,
        position: 'top-end',
        showConfirmButton: false,
        timer: 3000,
        timerProgressBar: true,
        didOpen: (toast) => {
          toast.addEventListener('mouseenter', Swal.stopTimer);
          toast.addEventListener('mouseleave', Swal.resumeTimer);
        },
      });

      Toast.fire({
        icon: 'error',
        title: 'Full cart!',
      });
    }
  };

  return (
    <Context.Provider value={{ data, handleValue, values, setArray, array }}>
      {children}
    </Context.Provider>
  );
};

export const useMyHookApplication = (): IContextApplication =>
  useContext(Context);
