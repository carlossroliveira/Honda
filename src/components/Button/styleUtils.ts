// -------------------------------------------------
// Packages
// -------------------------------------------------
import { css } from 'styled-components';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonVariantType } from './types';

const DEFAULT_STYLE = css`
  width: 172px;
  height: 28px;
`;

const NORMAL_STYLE = css`
  width: 165px;
  height: 44px;
`;

export const getStyled = (variant: ButtonVariantType) => {
  const Styled = {
    default: DEFAULT_STYLE,
    normal: NORMAL_STYLE,
  };
  return css`
    ${() => Styled[variant] || Styled.default}
  `;
};
