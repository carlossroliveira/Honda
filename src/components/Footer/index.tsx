// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ParagraphSC } from './footerStyles';

export const Footer = () => {
  const currentYear: number = new Date().getFullYear();

  return <ParagraphSC>&copy; {currentYear} | Carlos Oliveira</ParagraphSC>;
};
