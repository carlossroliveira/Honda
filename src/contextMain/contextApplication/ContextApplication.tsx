// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { createContext, useContext, useState } from 'react';
// -------------------------------------------------
// Hooks
// -------------------------------------------------
import { useFetch } from '../../hooks/useFetch';
// -------------------------------------------------
// Types
// -------------------------------------------------
import {
  IContextApplication,
  IContextApplicationProvider,
  IProductProps,
} from './types';

const Context = createContext({} as IContextApplication);

export const ThemeProviderApplication = ({
  children,
}: IContextApplicationProvider): JSX.Element => {
  const { data } = useFetch<IProductProps[]>('http://localhost:5000/info');

  const [array, setArray] = useState<any>([]);
  const [values, setValues] = useState<number>(0);

  if (values !== 4) {
    if (array.some((x: any) => array.indexOf(x) !== array.lastIndexOf(x))) {
      console.log('a');
    }
  }

  const handleValue = () => {
    if (values < 4) {
      setValues(values + 1);
    }
  };

  return (
    <Context.Provider value={{ data, handleValue, values, setArray, array }}>
      {children}
    </Context.Provider>
  );
};

export const useMyHookApplication = (): IContextApplication =>
  useContext(Context);
