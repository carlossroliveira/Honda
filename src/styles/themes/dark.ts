export default {
  title: 'dark',

  fontFamily: {
    fontDefault: 'Roboto, sans-serif',
  },
  color: {
    primary: '#ff8d66',
    secondary: '#393E46',
    tertiary: '',
    quaternary: '#F5F7FE',
  },
  background: {
    primary: '#F5F7FE',
    secondary: '#393E46',
    tertiary: '',
    quaternary: '#F5F7FE',
  },
  boxShadow: {
    primary:
      'rgba(50, 50, 93, 0.25) 0px 50px 100px -20px, rgba(0, 0, 0, 0.3) 0px 30px 60px -30px, rgba(10, 37, 64, 0.35) 0px -2px 6px 0px inset',
    secondary: '0px 2px 2px rgba(84, 89, 118, 0.46)',
  },
};
