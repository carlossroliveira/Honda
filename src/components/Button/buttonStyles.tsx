// -------------------------------------------------
// Packages
// -------------------------------------------------
import styled from 'styled-components';
import { getStyled } from './styleUtils';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonVariantType } from './types';

export const ButtonSC = styled.button<{
  variant: ButtonVariantType;
  shadow?: boolean;
}>`
  border: none;
  border-radius: 29px;
  transition: filter 0.2s;
  transition: 0.2s ease-in;

  color: ${({ theme }) => theme.color.secondary};
  box-shadow: ${({ shadow, theme }) => shadow && theme.boxShadow.primary};

  &:hover {
    filter: brightness(0.9);
    cursor: pointer;
  }

  ${({ variant }) => getStyled(variant)}
`;
