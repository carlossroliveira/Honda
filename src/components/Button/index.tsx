// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ButtonSC } from './buttonStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonProps } from './types';

export const Button = (props: ButtonProps) => {
  return (
    <ButtonSC ClassName={false} onClick={props.onClick} role="button">
      <>
        {props.text}
        {props.icon}
      </>
    </ButtonSC>
  );
};
