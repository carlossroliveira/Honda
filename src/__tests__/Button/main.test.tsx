// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { ThemeProvider } from 'styled-components';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Button } from '../../components/Button';
// -------------------------------------------------
// Context
// -------------------------------------------------
import { ThemeProviderApplication } from '../../contextMain/contextApplication/ContextApplication';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import dark from '../../styles/themes/dark';

describe('Button Testing', () => {
  it('Should render the component correctly.', () => {
    render(
      <ThemeProvider theme={dark}>
        <ThemeProviderApplication>
          <Button text={'Button'} variant={'default'} />
        </ThemeProviderApplication>
      </ThemeProvider>,
    );

    expect(
      screen.getByRole('button', {
        name: /button/i,
      }),
    ).toBeInTheDocument();
  });

  it('When clicking the button should be called correctly.', () => {
    const onClickTest = jest.fn();
    render(
      <ThemeProvider theme={dark}>
        <ThemeProviderApplication>
          <Button text={'Button'} onClick={onClickTest} variant={'default'} />
        </ThemeProviderApplication>
      </ThemeProvider>,
    );

    userEvent.click(
      screen.getByRole('button', {
        name: /button/i,
      }),
    );

    expect(onClickTest).toHaveBeenCalled();
    expect(onClickTest).toHaveBeenCalledTimes(1);
  });

  it('component rendering matches snapshot correctly', () => {
    render(
      <ThemeProvider theme={dark}>
        <ThemeProviderApplication>
          <Button text={'Button'} variant={'default'} />
        </ThemeProviderApplication>
      </ThemeProvider>,
    );
    expect(
      render(
        <ThemeProvider theme={dark}>
          <ThemeProviderApplication>
            <Button text={'Button'} variant={'default'} />
          </ThemeProviderApplication>
        </ThemeProvider>,
      ),
    ).toMatchSnapshot();
  });
});
