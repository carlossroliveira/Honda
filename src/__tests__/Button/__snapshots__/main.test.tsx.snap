// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 29px;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
  -webkit-transition: 0.2s ease-in;
  transition: 0.2s ease-in;
  color: #393E46;
  width: 172px;
  height: 28px;
  background-color: #F5F7FE;
}

.c0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
  cursor: pointer;
}

.c0 svg {
  margin-left: 5px;
}

<body>
    <div>
      <button
        class="c0"
        role="button"
      >
        Button
      </button>
    </div>
    <div>
      <button
        class="c0"
        role="button"
      >
        Button
      </button>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  border-radius: 29px;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
  -webkit-transition: 0.2s ease-in;
  transition: 0.2s ease-in;
  color: #393E46;
  width: 172px;
  height: 28px;
  background-color: #F5F7FE;
}

.c0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
  cursor: pointer;
}

.c0 svg {
  margin-left: 5px;
}

<div>
    <button
      class="c0"
      role="button"
    >
      Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
